sklearn/cluster/_bisect_k_means.py

```diff
@@ -0,0 +1,727 @@
+                # to pick cluster with largest number of data points assigned
+                metrics_values = np.bincount(_labels)
+
+            # "Create Hierarchy":
+            # Cluster with smaller metrics value (SSE or ammount of points)
+            # will be at 'left side' and cluster with higher at 'right side'
+            lower_id = 0 if metrics_values[0] <= metrics_values[1] else 1
+            centers_id = [lower_id, 1 - lower_id]
+
+            # Assign calculated nested clusters to their root
+            tree_dict[biggest_id]["children"] = [last_center_id, last_center_id + 1]
+
+            for id in range(2):
+                child_id = tree_dict[biggest_id]["children"][id]
+                picked_id = centers_id[id]
+
+                # Save Results on Tree
+                tree_dict[child_id] = {"children": None, "center": _centers[picked_id]}
+
+                # Create Mask for samples for selecting proper data points
+                samples_mask = leaves_dict[biggest_id]["samples"].copy()
+                samples_mask[picked_samples] = _labels == picked_id
+
+                # Save recently generated leaves
+                leaves_dict[child_id] = {
+                    "samples": samples_mask,
+                    "error_or_size": metrics_values[picked_id],
+                }
+
+            # Split cluster is no longer leaf
+            del leaves_dict[biggest_id]
+
+            last_center_id += 2
+
+            # Pick new 'biggest cluster' to split
+            biggest_id, _ = max(
+                leaves_dict.items(), key=lambda x: x[1]["error_or_size"]
+            )
+
+        # Delete Initial cluster
+        del tree_dict[-1]
+
+        # Sort Leaf centers by their root
+        centers_hierarchical = [0, 1]
+        while len(centers_hierarchical) != self.n_clusters:
+            nested_centers = []
+
+            for center_id in centers_hierarchical:
+                center_ids = (
+                    [center_id]
+                    if tree_dict[center_id]["children"] is None
+                    else tree_dict[center_id]["children"]
+                )
+                nested_centers.extend(center_ids)
+
+            centers_hierarchical = nested_centers
+
+        # Initialize Labels
+        labels = np.full(X.shape[0], -1, dtype=np.intc)
+
+        centers = []
+
+        for i, center_id in enumerate(centers_hierarchical):
+            # Save cluster centers in hierarchical order
+            centers.append(tree_dict[center_id]["center"])
+
+            # Assign labels to proper data points
+            labels[leaves_dict[center_id]["samples"]] = i
+
+        centers = np.asarray(centers)
+
+        # Inner Tree will be later used at 'predict' method:
+        self._inner_tree = tree_dict
+
+        return centers, labels
+
+    def predict(self, X, sample_weight=None):
+        """Predict which cluster each sample in X belongs to by going down
```

Comment:
The short docstring description should fit in one line.
Please move the details to a separate paragraph in the docstring.

Reply:
Changed